pkg/auth/auth.go

```diff
@@ -147,24 +167,68 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		return nil
 	}
 	if unauthorized, ok := err.(docker.ErrUnauthorizedForCredentials); ok {
-		logrus.Debugf("error logging into %q: %v", server, unauthorized)
-		return errors.Errorf("error logging into %q: invalid username/password", server)
+		logrus.Debugf("error logging into %q: %v", key, unauthorized)
+		return errors.Errorf("error logging into %q: invalid username/password", key)
 	}
-	return errors.Wrapf(err, "authenticating creds for %q", server)
+	return errors.Wrapf(err, "authenticating creds for %q", key)
+}
+
+// parseRegistryArgument verifies the provided arg depending if we accept
+// repositories or not.
+func parseRegistryArgument(arg string, acceptRepositories bool) (key, registry string, maybeRef reference.Named, err error) {
+	if acceptRepositories {
+		key = trimScheme(arg)
+		if key != arg {
+			err = errors.New("credentials key has https[s]:// prefix")
+			return
+		}
+
+		registry = getRegistryName(key)
+		if registry == key {
+			// We cannot parse a reference from a registry, so we stop here
+			return
```

Comment:
(Non-blocking personal preference: I would find all of this easier to read with explicit `return key, registry, nil, nil`, and the like.

OTOH it does work perfectly fine.)

Reply:
Fixed :+1: